### DISCUSSION TITLE
gmake on linux

### DISCUSSION THREAD

**Discussion by mconfalonieri** [Homebrew/discussions#1787] [Everyday usage]
Hi all!

I am fairly new to Homebrew. I am trying to create a tap for my program sadplay, to ease installation, but I faced different problems.

The last one is that my software passes the test-bot for MacOS, but not for Linux. I cannot access the logs, but I am wondering if the cause could be that in my formula I am using `gmake`, since on Linux it is usually simply `make`:

```ruby
system "gmake", "install"
```

If this is the case, is it possible to specify a different command for each architecture?

---

For reference, here is the link to the failed test:

https://github.com/mconfalonieri/homebrew-sadplay/pull/4/checks?check_run_id=2994019489

**Comment by carlocab** [comment]
Have you tried changing it to
```ruby
system "make", "install"
```
?

**Reply by mconfalonieri** [reply]
I will do a quick test, thank you for the reply!

**Reply by mconfalonieri** [reply]
So, by using `make` I could build on both system. However, on Linux it failed anyway because it could not find a package called `libgpg-error`:

```
==> Downloading https://ghcr.io/v2/linuxbrew/core/libgpg-error/blobs/sha256:b862b10316313dabc9ddc7774e37c41c0700b886d361b86a1485052287b73fd0
170
curl: (22) The requested URL returned error: 404 
171
Error: Failed to download resource "libgpg-error"
```

I modified the test workflow to install the package with the `--build-bottle` parameter. I did it already for another dependency, `adplug`, as it seems that there are no bottles available for Linux for this package.

Now it works!
